kore/src/Kore/Step/Axiom/Matcher.hs

```diff
@@ -895,3 +897,20 @@ keyBasedIntersectionDifference merger firsts seconds =
         result@IntersectionDifference{intersection}
         (These first second)
       = result {intersection = merger first second : intersection}
+
+{- | Renormalize builtin types after substitution.
+ -}
+renormalizeBuiltins
```

Comment:
this was moved here to avoid an import cycle

Reply:
```
Module imports form a cycle:
         module ‘Kore.Builtin’ (/home/traian/Documents/kore/kore/src/Kore/Builtin.hs)
        imports ‘Kore.Builtin.KEqual’ (/home/traian/Documents/kore/kore/src/Kore/Builtin/KEqual.hs)
  which imports ‘Kore.Step.Simplification.And’ (/home/traian/Documents/kore/kore/src/Kore/Step/Simplification/And.hs)
  which imports ‘Kore.Step.Simplification.AndTerms’ (/home/traian/Documents/kore/kore/src/Kore/Step/Simplification/AndTerms.hs)
  which imports ‘Kore.Step.Simplification.Exists’ (/home/traian/Documents/kore/kore/src/Kore/Step/Simplification/Exists.hs)
  which imports ‘Kore.Step.Axiom.Matcher’ (/home/traian/Documents/kore/kore/src/Kore/Step/Axiom/Matcher.hs)
  which imports ‘Kore.Builtin’ (/home/traian/Documents/kore/kore/src/Kore/Builtin.hs)
```

Reply:
OK, I think this change is fine to avoid that cycle. Eventually, we will break the cycle between `Kore.Step.Simplification.AndTerms` and `Kore.Step.Simplification.Exists`.